server/encoder/video_encoder_nvenc.cpp

```diff
@@ -80,28 +88,46 @@ static auto encode_guid(video_codec codec)
 	throw std::out_of_range("Invalid codec " + std::to_string(codec));
 }
 
-static void check_encode_guid_supported(std::shared_ptr<video_encoder_nvenc_shared_state> shared_state, void * session_handle, GUID encodeGUID)
+static void assert_encode_guid_supported(std::shared_ptr<video_encoder_nvenc_shared_state> shared_state, void * session_handle, GUID encodeGUID)
 {
 	uint32_t count;
-	std::vector<GUID> encodeGUIDs;
 	NVENC_CHECK(shared_state->fn.nvEncGetEncodeGUIDCount(session_handle, &count));
-	encodeGUIDs.resize(count);
+
+	std::vector<GUID> encodeGUIDs(count);
 	NVENC_CHECK(shared_state->fn.nvEncGetEncodeGUIDs(session_handle, encodeGUIDs.data(), count, &count));
 
-	for (GUID guid: encodeGUIDs)
+	if (!std::ranges::contains(encodeGUIDs, encodeGUID))
 	{
-		if (guid.Data1 == encodeGUID.Data1 &&
-		    guid.Data2 == encodeGUID.Data2 &&
-		    guid.Data3 == encodeGUID.Data3)
-		{
-			if (std::equal(std::begin(guid.Data4), std::end(guid.Data4), std::begin(encodeGUID.Data4)))
-			{
-				return;
-			}
-		}
+		throw std::runtime_error("nvenc: GPU doesn't support selected codec.");
+	}
+}
+
+static void assert_preset_guid_supported(std::shared_ptr<video_encoder_nvenc_shared_state> shared_state, void * session_handle, GUID encodeGUID, GUID presetGUID)
```

Comment:
I would prefer `check_preset_guid_supported`, as we expect assertion to be the `assert` macro, which depends on compile type.